dove/tests/test_cmd_dove_init.rs

```diff
@@ -0,0 +1,347 @@
+#![cfg(test)]
+
+use std::fs::{read_to_string, create_dir_all};
+use toml::Value;
+
+mod test_cmd_helper;
+use crate::test_cmd_helper::{
+    project_start, project_start_for_init, project_remove, set_dependencies_local_move_stdlib,
+    project_build, execute_dove_at, execute_dove_at_wait_fail,
+};
+
+/// $ dove init
+#[test]
+fn test_cmd_dove_init_without_arguments() {
+    // Project name and path
+    let project_name = "demoproject_36";
+    let project_folder = project_start_for_init(project_name);
+
+    execute_dove_at(&project_folder, &["dove", "init"]);
+    // Check config
+    let package = read_to_string(project_folder.join("Dove.toml"))
+        .unwrap()
+        .parse::<Value>()
+        .unwrap()
+        .get("package")
+        .unwrap()
+        .clone();
+
+    assert!(
+        package
+            .get("name")
+            .unwrap()
+            .to_string()
+            .contains(project_name),
+        "Dove.toml: invalid name",
+    );
+
+    assert!(
+        package.get("dialect").unwrap().to_string().contains("pont"),
+        "Dove.toml: invalid dialect",
+    );
+
+    set_dependencies_local_move_stdlib(&project_folder);
+    project_build(&project_folder);
+    project_remove(&project_folder);
+}
+/// $ dove init -d ###
+#[test]
+fn test_cmd_dove_init_with_dialect() {
+    // Project name and path
+    let project_name = "demoproject_43";
+    let project_folder = project_start_for_init(project_name);
+
+    for dialect in &["pont", "diem", "dfinance"] {
+        execute_dove_at(&project_folder, &["dove", "init", "-d", dialect]);
+        // Check config
+        let package = read_to_string(project_folder.join("Dove.toml"))
+            .unwrap()
+            .parse::<Value>()
+            .unwrap()
+            .get("package")
+            .unwrap()
+            .clone();
+
+        assert!(
+            package
+                .get("name")
+                .unwrap()
+                .to_string()
+                .contains(project_name),
+            "Dove.toml: invalid name",
+        );
+
+        assert!(
+            package
+                .get("dialect")
+                .unwrap()
+                .to_string()
+                .contains(dialect),
+            "Dove.toml: invalid dialect",
+        );
+
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder);
```

Comment:
I assume it's here to check whether project will still be able to build after `dove init`? Better make this with explicit assert, and move to `.is_ok()` function suggested by @vitvakatu